src/main/java/com/urswolfer/gerrit/client/rest/http/GerritRestClient.java

```diff
@@ -274,11 +276,13 @@ private void setRequestBody(String requestBody, HttpRequestBase method, ContentT
             cookieStore.clear();
             return Optional.absent();
         }
-        if (authData.isHttpPassword()) {
+        if (authData.isHttpPassword() || loginCache.isGithubOAuthDetected()) {
```

Comment:
I'd split out this 2nd condition into a seprate if, and move the comment into that (then we are safe that the comment always stays where it belongs, even when we move around things:

```
if (loginCache.isGithubOAuthDetected()) {
  // Also on GitHub/OAuth authentication, do not keep on trying the /login
  // page as it would just result in a continuous loop of failed login attempts.
  return Optional.absent();
}
```

Reply:
Good idea, as it would make it more readable and less error-prone. 👍